pylint/testutils/functional/find_functional_tests.py

```diff
@@ -38,9 +38,11 @@ def get_functional_test_files_from_directory(
 
     _check_functional_tests_structure(Path(input_dir))
 
-    for dirpath, _, filenames in os.walk(input_dir):
+    for dirpath, dirnames, filenames in os.walk(input_dir):
         if dirpath.endswith("__pycache__"):
             continue
+        dirnames.sort()
+        filenames.sort()
```

Comment:
```suggestion
```
Why does ``dirnames`` need to be sorted ? We're not using it afaik ?

Reply:
This is the [recommended](https://docs.python.org/3/library/os.html#:~:text=the%20caller%20can%20modify%20the%20dirnames%20list%20in%2Dplace) way to make the directory iteration order deterministic using `os.walk()`.

Reply:
Relevant part of the doc:

> When topdown is True, the caller can modify the dirnames list in-place (perhaps using [del](https://docs.python.org/3/reference/simple_stmts.html#del) or slice assignment), and [walk()](https://docs.python.org/3/library/os.html#os.walk) will only recurse into the subdirectories whose names remain in dirnames; this can be used to prune the search, impose a specific order of visiting, or even to inform [walk()](https://docs.python.org/3/library/os.html#os.walk) about directories the caller creates or renames before it resumes [walk()](https://docs.python.org/3/library/os.html#os.walk) again. Modifying dirnames when topdown is False has no effect on the behavior of the walk, because in bottom-up mode the directories in dirnames are generated before dirpath itself is generated.